### PR TITLE
Implemented gps location into map.

### PR DESCRIPTION
**Pull request by Arthelh**
Implemented the gps: A new button is there on the map, when clicked on it brings the map back to the user position. Tested on my phone and it did bring me back to my current location.
Apparently it does not even need to be tested which is beatiful (I do not see the new code anywhere in the jacoco index file.)

### CODE REVIEW

**Review by Jnf14 — APPROVED** [review]
Nothing to say but good job ! Nice feature to have and the implementation will help us on live tracking !

File: app/src/main/java/com/ncnf/map/ui/MapFragment.java [hunk 1]
```diff
@@ -58,13 +66,21 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
 
         View view = inflater.inflate(R.layout.fragment_map, container, false);
 
+        requestPermissionLauncher = registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
+            if (isGranted) {
+                returnToLocation(view);
+            }
+        });
+
```
Comment (khcr) [inline app/src/main/java/com/ncnf/map/ui/MapFragment.java:74]: What happens if the user refuses the permission ?

**Review by khcr — APPROVED** [review]
A great self-contained new feature. No new tests added, should the button click be tested (even if the coverage is not impacted) ?

File: app/src/main/java/com/ncnf/map/ui/MapFragment.java (hunk 1 above)
Comment (Arthelh) [reply app/src/main/java/com/ncnf/map/ui/MapFragment.java:74]: Nothing, the function is not called and the position remains the same

**Comment by Arthelh** [comment]
> A great self-contained new feature. No new tests added, should the button click be tested (even if the coverage is not impacted) ?

I was thinking of it, problem is the only thing the button does is modify the position and then update the map accordingly. Unlike most other buttons, no intent or any other thing I know how to test is created. I also checked if maybe I could test that the popup shows for the permission, but when researching permissions and Espresso, I only found answers that told how to make so that the permission is given to Espresso, nothing about the popup itself.

**Pull request merged**